Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp

```diff
@@ -355,13 +355,15 @@ void SourceBufferPrivateRemote::setAppendWindowEnd(const MediaTime& appendWindow
     gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::SetAppendWindowEnd(appendWindowEnd), m_remoteSourceBufferIdentifier);
 }
 
-void SourceBufferPrivateRemote::seekToTime(const MediaTime& mediaTime)
+void SourceBufferPrivateRemote::seekToTarget(const WebCore::SeekTarget& target, CompletionHandler<void(const MediaTime&)>&& completionHandler)
 {
     auto gpuProcessConnection = m_gpuProcessConnection.get();
-    if (!isGPURunning())
+    if (!isGPURunning()) {
+        completionHandler(MediaTime::invalidTime());
         return;
+    }
 
-    gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::SeekToTime(mediaTime), m_remoteSourceBufferIdentifier);
+    gpuProcessConnection->connection().sendWithAsyncReply(Messages::RemoteSourceBufferProxy::SeekToTarget(target), WTFMove(completionHandler), m_remoteSourceBufferIdentifier);
```

Comment:
Ditto, invalidTime is probably what we want if GPUProcess crashes within the async call.

Reply:
I've created a `AsyncReplyError<WTF::MediaTime>` that sets the default return value to `MediaTime::invalidTime()`

thanks for the pointers